Source/HTTP/httpcall.cpp

```diff
@@ -449,6 +449,11 @@ try
     retryContext->outerAsyncBlock = asyncBlock;
     retryContext->outerQueue = asyncBlock->queue;
     retry_context* rawRetryContext = static_cast<retry_context*>(shared_ptr_cache::store<retry_context>(retryContext));
+    if (rawRetryContext == nullptr)
+    {
+        XAsyncComplete(asyncBlock, E_HC_NOT_INITIALISED, 0);
+        return E_HC_NOT_INITIALISED;
+    }
```

Comment:
maybe try to get_http_singleton at the beginning of the function and return E_NOT_INITIALIZED before duplicating the handle/allocating the context? I don't think you need to complete the async block in this case either.

Reply:
Done